SpatialGDK/Source/SpatialGDK/Private/EngineClasses/GridBasedLBStrategy.cpp

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "EngineClasses/GridBasedLBStrategy.h"
+
+#include "EngineClasses/SpatialNetDriver.h"
+#include "Utils/SpatialActorUtils.h"
+
+UGridBasedLBStrategy::UGridBasedLBStrategy()
+	: Super()
+	, Rows(1)
+	, Cols(1)
+	, WorldWidth(10000.f)
+	, WorldHeight(10000.f)
+{
+}
+
+void UGridBasedLBStrategy::Init(const USpatialNetDriver* InNetDriver)
+{
+	Super::Init(InNetDriver);
+
+	for (uint32 i = 1; i <= Rows * Cols; i++)
+	{
+		VirtualWorkerIds.Add(i);
+	}
+
+	const float WorldWidthMin = -(WorldWidth / 2.f);
+	const float WorldHeightMin = -(WorldHeight / 2.f);
+
+	const float ColumnWidth = WorldWidth / Cols;
+	const float RowHeight = WorldHeight / Rows;
+
+	float XMin = WorldWidthMin;
+	float YMin = WorldHeightMin;
+	float XMax, YMax;
+
+	for (uint32 Col = 0; Col < Cols; ++Col)
+	{
+		XMax = XMin + ColumnWidth;
```

Comment:
If we want to be super strict about this we probably need to make sure that the last one is the world extent. I'm not sure it really matters though.

Reply:
I think at this point it's up to the use to specify the correct world size. Users might cut themselves on this a lot though, so in the future we could extend the edge cells to infinity, or warn whenever something is outside of all cells.

The warning might be something we do at the GDK level, warning when an actor was mapped to the invalid virtual worker id.